lib/themes/dosomething/paraneue_dosomething/js/takeover/Takeover.js

```diff
@@ -0,0 +1,245 @@
+
+    if (slide) {
+      this.takeoverContent.hide();
+      this.takeoverContent.delay(500).slideDown('slow', function() {});
+    }
+  },
+
+  /**
+   * Binds click handlers to $button that sends an AJAX request and uses deferred promises to
+   * run callbacks in the order in which they are recieved.
+   *
+   * @param  {jQuery}   $button   Signup buttons
+   * @param  {object}   request   An object that defines the properties we need to make a request.
+   */
+  handleSubmit: function($button, request) {
+    const _this = this;
+
+    // Create custom deferred promise object.
+    let ajaxPromise = $.Deferred().resolve().promise();
+
+    // Bind click handler
+    $button.click(function(e) {
+      e.preventDefault();
+
+      const $this = $(this);
+
+      if (! $('.validation__message.has-error').length) {
+        // Make the ajax call, store the promise.
+        const promise = _this.sendRequest(request);
+
+        // Fire done call backs in the order in which they were recieved.
+        ajaxPromise = ajaxPromise.then(function() {
+          return promise;
+        }).done(function(data) {
+          _this.changeScreens('next');
+        });
+      }
+    });
+  },
+
+  /**
+   * Sends an AJAX request using the provided information in the request object that is passed in.
+   *
+   * @param  {request}   The object containing the details about the request i.e URL, method, data
+   * @return {object}   The api response.
+   */
+  sendRequest: function(request) {
+    const _this = this;
+
+    return $.ajax({
+      method: request.method,
+      url: request.url,
+      dataType: 'json',
+      headers: {
+        'Accepts': 'application/json',
+        'Content-Type': 'application/json',
+        'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content'),
+      },
+      data: request.data,
+      beforeSend: function() {
+        _this.addSpinner(request.button);
+      },
+    });
+  },
+
+  /*
+   * Store the indexes of each of the screens provided in the template.
+   */
+  setScreenIndexes: function() {
+    let indexes = [];
+
+    $.each(this.screens, function(scr) {
+      indexes.push($(scr).data('index'));
+    });
+
+    return indexes;
+  },
+
+  /**
+   * Handles makeing a new scren active based on the direction passed.
+   *
+   * @param  {string}   direction
+   */
+  changeScreens: function(direction) {
+    // Check if an allowed direction was passed in.
+    if ($.inArray(direction, ['next', 'previous']) === -1) {
+      return;
+    }
+
+    const $current = $('.takeover__screen.active');
+    const currentIndex = $current.data('index');
+    const newIndex = (direction === 'next') ? currentIndex + 1 : currentIndex - 1;
+
+    // Check if the new screen exists, if so, go there, otherwise we need to loop back.
+    if ($.inArray(newIndex, this.screenIndexes)) {
+      $current.removeClass('active');
+      $('.takeover__screen[data-index="' + newIndex + '"]').addClass('active');
```

Comment:
This'd be a lovely place to use a [template string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals):
```js
$(`.takeover__screen[data-index="${newIndex}"]`)
```